### PR TITLE
[Tests] Improve unet / vae tests

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Add aggressive unet, vae single-forward tests with 1e-4 precision.

Using parameterized tests over seeds and timesteps.

@anton-l I'd say we see whether precision holds over the next days (weekend) and then we eventually adapt next week.

@patil-suraj @pcuenca @anton-l - note that tests are now automatically suffixed with `_{0, 1, 2, 4}` when using `parameterized` => this makes it slightly more unreadable/useable when running locally, e.g.:
```
tests/models/test_models_unet_2d.py::UNet2DConditionModelIntegrationTests::test_compvis_sd_inpaint_0 PASSED
tests/models/test_models_unet_2d.py::UNet2DConditionModelIntegrationTests::test_compvis_sd_inpaint_1 PASSED
tests/models/test_models_unet_2d.py::UNet2DConditionModelIntegrationTests::test_compvis_sd_inpaint_2 PASSED
tests/models/test_models_unet_2d.py::UNet2DConditionModelIntegrationTests::test_compvis_sd_inpaint_3 PASSED
```

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Slow tests currently don't work on CPU because of: https://github.com/huggingface/accelerate/issues/796 

Think that's fine as they are only run on GPU anyways - device_map="auto" speeds up init from ~25seconds to ~3seconds so this does make a big difference for overall testing - cc @anton-l

File: 1 [hunk 1]
```diff
@@ -19,9 +19,10 @@
```
Comment (anton-l) [inline 1:19]: Looks like accidental renaming here, `tests/test_models_vae.py → 1`

File: tests/models/test_models_unet_2d.py [hunk 2]
```diff
@@ -397,7 +355,7 @@ def test_from_pretrained_hub(self):
 
     @slow
     def test_output_pretrained_ve_mid(self):
-        model = UNet2DModel.from_pretrained("google/ncsnpp-celebahq-256")
+        model = UNet2DModel.from_pretrained("google/ncsnpp-celebahq-256", device_map="auto")
```
Comment (anton-l) [inline tests/models/test_models_unet_2d.py:358]: Nice to see `"auto"` helping the latency!

File: tests/models/test_models_unet_2d.py [hunk 3]
```diff
@@ -449,3 +407,194 @@ def test_output_pretrained_ve_large(self):
...
+            [83, 4, [-0.2323, -0.1304, 0.0813, -0.3093, -0.0919, -0.1571, -0.1125, -0.5806]],
+            [17, 0.55, [-0.0831, -0.2443, 0.0901, -0.0919, 0.3396, 0.0103, -0.3743, 0.0701]],
+            [8, 0.89, [-0.4863, 0.0859, 0.0875, -0.1658, 0.9199, -0.0114, 0.4839, 0.4639]],
+            [3, 1000, [-0.5649, 0.2402, -0.5518, 0.1248, 1.1328, -0.2443, -0.0325, -1.0078]],
+            # fmt: on
+        ]
+    )
+    @require_torch_gpu
+    def test_compvis_sd_v1_4_fp16(self, seed, timestep, expected_slice):
+        pass
```
Comment (anton-l) [inline tests/models/test_models_unet_2d.py:481]: Is it meant to be skipped for now? 4 more passes in the test below too

**Review by anton-l — APPROVED** [review]
Really like this solution, thank you! I'll run these on different machines once merged, to hopefully add some datapoints for us

File: tests/models/test_models_unet_2d.py (hunk 3 above)
Comment (patrickvonplaten) [reply tests/models/test_models_unet_2d.py:481]: Ups

File: 1 (hunk 1 above)
Comment (patrickvonplaten) [reply 1]: Thanks!

**Pull request merged**